ci/Dockerfile.ledger

```diff
@@ -1,4 +1,4 @@
-FROM ghcr.io/ledgerhq/speculos
+FROM ghcr.io/ledgerhq/speculos:sha-f430e51
```

Comment:
@notmandatory Any reason for using this specific hash? As per my tests, the most recent version that's working properly is 0.8.3.

Reply:
no reason I just picked one I thought was around the time we last had this working. If you have it working with 0.8.3 then let's use that one. I updated this PR to test it out too and 0.8.3 looks fine.